core/src/test/java/org/elasticsearch/index/analysis/WhitespaceTokenizerFactoryTests.java

```diff
@@ -0,0 +1,83 @@
+
+import com.carrotsearch.randomizedtesting.generators.RandomStrings;
+
+import org.apache.lucene.analysis.core.WhitespaceTokenizer;
+import org.elasticsearch.Version;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.common.settings.Settings;
+import org.elasticsearch.index.Index;
+import org.elasticsearch.index.IndexSettings;
+import org.elasticsearch.test.ESTestCase;
+import org.elasticsearch.test.IndexSettingsModule;
+
+import java.io.IOException;
+import java.io.Reader;
+import java.io.StringReader;
+
+import static org.apache.lucene.analysis.BaseTokenStreamTestCase.assertTokenStreamContents;
+
+public class WhitespaceTokenizerFactoryTests extends ESTestCase {
+
+    public void testSimpleWhiteSpaceTokenizer() throws IOException {
+        final Settings indexSettings = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build();
+        IndexSettings indexProperties = IndexSettingsModule.newIndexSettings(new Index("test", "_na_"), indexSettings);
+        WhitespaceTokenizer tokenizer = (WhitespaceTokenizer) new WhitespaceTokenizerFactory(indexProperties, null, "whitespace_maxlen",
+                Settings.EMPTY).create();
+
+        try (Reader reader = new StringReader("one, two, three")) {
+            tokenizer.setReader(reader);
+            assertTokenStreamContents(tokenizer, new String[] { "one,", "two,", "three" });
+        }
+    }
+
+    public void testMaxTokenLength() throws IOException {
+        final Settings indexSettings = Settings.builder().put(IndexMetaData.SETTING_VERSION_CREATED, Version.CURRENT).build();
+        IndexSettings indexProperties = IndexSettingsModule.newIndexSettings(new Index("test", "_na_"), indexSettings);
+        final Settings settings = Settings.builder().put(WhitespaceTokenizerFactory.MAX_TOKEN_LENGTH, 2).build();
+        WhitespaceTokenizer tokenizer = (WhitespaceTokenizer) new WhitespaceTokenizerFactory(indexProperties, null, "whitespace_maxlen",
+                settings).create();
+        try (Reader reader = new StringReader("one, two, three")) {
+            tokenizer.setReader(reader);
+            assertTokenStreamContents(tokenizer, new String[] { "on", "e,", "tw", "o,", "th", "re", "e" });
+        }
+
+        final Settings defaultSettings = Settings.EMPTY;
+        tokenizer = (WhitespaceTokenizer) new WhitespaceTokenizerFactory(indexProperties, null, "whitespace_maxlen", defaultSettings)
+                .create();
+        String veryLongToken = RandomStrings.randomAsciiAlphanumOfLength(random(), 256);
+        try (Reader reader = new StringReader(veryLongToken)) {
+            tokenizer.setReader(reader);
+            assertTokenStreamContents(tokenizer, new String[] { veryLongToken.substring(0, 255), veryLongToken.substring(255) });
+        }
+
+        final Settings tooLongSettings = Settings.builder().put(WhitespaceTokenizerFactory.MAX_TOKEN_LENGTH, 1024 * 1024 + 1).build();
```

Comment:
@jpountz from your comment in #26643 I think you might find this bound to high, but I decided to start with this since its also the one that e.g. the standard tokenizer uses and the bound is set in Lucene. Might be worth lowering, but then again having the same max value in all the basic tokenizers might be more consistent, wdyt?

Reply:
+1